### ISSUE TITLE
HELP: looking for new maintainers

### ISSUE THREAD

**Issue by saghul** [saghul/pycares#139]
TLDR: I can no longer maintain this  project to the degree  it needs to be maintained and I'm looking for a new maintainer(s). If you'd like to step up or propose a way forward (someone mentioned the aio-libs org) please leave a comment.  ❤️ 

<hr />

I started this project almost 10 years ago, in my quest to improve my C skills and learn more about Python internals. In this process I also learned about how c-ares works and many more things. It started with the Python C API, and then, thanks to multiple contributions we transitioned to cffi. Then came wheels.

Python is and has always been my favorite programing language, but I no longer use it on a daily basis and my day got more complicated so I no longer have the time to pour some love into this project. I never had a use case for this package, I just did it for the sake of learning. Python being my favorite language, I loved to come back to fix the occasional bug, but that is no longer working.

I've come to realize there is a grieving process here. It starts with denial: "I'll find some time next month for sure!" and ends with acceptance, ie, this piece of text right here.

This project has helped me become a better developer and to collaborate with many great people throughout the years. It was really humbling to find out that `aiohttp` started depending on it, and that's probably how it reacched [750k downloads per month](https://twitter.com/hynek/status/1359581079640932361).

It's not time to pass the torch and put this project in new, loving hands. Are you that person? Please leave a comment.

**Comment by hynek** [comment]
The real source btw: https://pepy.tech/project/pycares :) Since aiodns is very close, you're probably right: https://pepy.tech/project/aiodns

**Comment by kigawas** [comment]
[c-ares](https://github.com/c-ares/c-ares) is evolving quickly, but this project is still using an ancient c-ares version, which causes lots of build problems with Python 3.9 on Ubuntu 20.04 and macOS 11 due to:

> Upstream c-ares is not willing to apply a patch adding TTL support. I did apply the patch to the bundled c-ares, but unfortunately it breaks the ABI, so attempting to use a system provided c-ares is not possible.

In my opinion, c-ares has some TTL support differently now. I'm not a DNS expert, but it's really doubted that many would need "TTL in SOA, TXT, SRV and MX". If they do, it should have been integrated into c-ares long time ago.

If you really want to cease maintaining, I suggest two options:

1. Drop the pycares dependencies for aiodns and use ffi directly with c-ares in aiodns (requiring some work)
2. Upgrade the c-ares code and apply your patch to fix build errors on latest *nix OSs (requiring less work)

For long term development, 1 is better. If we want to get things fixed ASAP, 2 is also a good start.

**Comment by boytm** [comment]
I can try it， but English is not my native language.

**Comment by webknjaz** [comment]
Sounds like this lib and aiodns are indeed perfect candidates for @jazzband (cc @jezdez). It's dramatically easier to maintain things collectively + this ensures there are fewer single points of failure if somebody wants to step down from contributing regularly.

One thing that would be helpful for the new maintainers is some docs around what is the canonical package build workflow and test runtime considerations. If that's missing, I envision it to be hard for somebody new to figure out the processes — having the need to do Git archeology substantially increases the entry level for the potential future maintainers.

**Comment by HassanAbouelela** [comment]
Not sure if much is still happening here, but I needed a 3.9 wheel, so I just went ahead and made it on my fork and PRed it. Some work is still needed to publish it to pypi, but I've done most of the grunt work in bumping versions, updating CI, and documentation. I've tested the package a little bit in my own projects, and nothing seems to be out of the ordinary.

Pypi currently lists just @saghul as the only maintainer, I hope they have a chance to look at my PR soon so we can get the ball rolling again.

**Comment by saghul** [comment]
Will do, thanks a lot for picking this up!

**Comment by saghul** [comment]
Update 2: got excited with all the automation and made a 4.0 release. c-ares is now pulled in as a submodule. No custom patches. Linking with the system c-ares is now possible too.

With these improvements and the release automation it should be much easier for anyone to come and help maintain this forward!

**Comment by Dreamsorcerer** [comment]
@saghul As this is a dependency of aiodns and we need to keep it maintained, we could take this on under @aio-libs if you're still looking for help.

**Comment by saghul** [comment]
I'd be happy to do that as I already did for aiodns. I'd love to see some PRs come in first though.

**Comment by Dreamsorcerer** [comment]
I'm working on improving the type checking at the moment, but looks like I'll need to do typing improvements here first.

**Comment by saghul** [comment]
@bdraco Thanks for all the work you put in the latest release! Would you like the commit bit?

**Comment by bdraco** [comment]
> [@bdraco](https://github.com/bdraco) Thanks for all the work you put in the latest release! Would you like the commit bit?

Yes, thank you, and I appreciate your help as well. That would be great; I'm happy to continue helping out going forward!

**Comment by saghul** [comment]
Just invited you!

**Comment by bdraco** [comment]
Thanks. Are you ok with me setting up dependabot to keep the CI up to date?  I'd also like to switch everything over to trusted publishing as well. You should be able to add it on https://pypi.org/manage/project/pycares/settings/publishing/

**Comment by saghul** [comment]
Sure thing!

**Comment by bdraco** [comment]
@saghul Would you please set up security reporting on this repo. We have a report that needs to be shared privately.

**Comment by webknjaz** [comment]
@saghul there's a toggle in the settings for allowing submissions by external users specifically. This is what we have enabled across aio-libs and it's a very convenient private collaboration interface.

**Comment by saghul** [comment]
Sorry, looking into this now.

**Comment by saghul** [comment]
Done!

**Comment by bdraco** [comment]
The security issue ended up getting opened on `aiodns` so you should be able to see it there.

I'm being light on details intentionally because of the sensitive nature of security issues

**Comment by webknjaz** [comment]
@saghul above you mentioned wanting to move this under @aio-libs at some point. Is that still the case?

**Comment by saghul** [comment]
This package supports more an asyncio and it should be in good shape for now, thanks for checking in!
